waku/v2/peermanager/topic_event_handler_test.go

```diff
@@ -0,0 +1,297 @@
+	pm.ctx = ctx
+	go pm.connectivityLoop(ctx)
+
+	// Subscribe to Pubsub topic
+	_, err = r.Subscribe(ctx, protocol.NewContentFilter(pubSubTopic))
+	require.NoError(t, err)
+
+	// Call the appropriate handler
+	select {
+	case e := <-pm.sub.Out():
+		switch e := e.(type) {
+		case relay.EvtRelaySubscribed:
+			{
+				eventDetails := (relay.EvtRelaySubscribed)(e)
+				pm.handleNewRelayTopicSubscription(eventDetails.Topic, eventDetails.TopicInst)
+			}
+		default:
+			require.Fail(t, "unexpected event arrived")
+		}
+
+	case <-ctx.Done():
+		require.Fail(t, "closed channel")
+	}
+
+	// Check Peer Manager knows about the topic
+	_, ok := pm.subRelayTopics[pubSubTopic]
+	require.True(t, ok)
+
+	// UnSubscribe from Pubsub topic
+	err = r.Unsubscribe(ctx, protocol.NewContentFilter(pubSubTopic))
+	require.NoError(t, err)
+
+	// Call the appropriate handler
+	select {
+	case e := <-pm.sub.Out():
+		switch e := e.(type) {
+		case relay.EvtRelayUnsubscribed:
+			{
+				eventDetails := (relay.EvtRelayUnsubscribed)(e)
+				pm.handleNewRelayTopicUnSubscription(eventDetails.Topic)
+			}
+		default:
+			require.Fail(t, "unexpected event arrived")
+		}
+
+	case <-ctx.Done():
+		require.Fail(t, "closed channel")
+	}
+
+	// Check the original topic was removed from Peer Manager
+	_, ok = pm.subRelayTopics[pubSubTopic]
+	require.False(t, ok)
+
+}
+
+func TestHandlePeerTopicEvent(t *testing.T) {
+	log := utils.Logger()
+	pubSubTopic := "/waku/2/go/pm/test"
+	ctx := context.Background()
+
+	hosts := make([]host.Host, 5)
+	relays := make([]*relay.WakuRelay, 5)
+
+	for i := 0; i < 5; i++ {
+		relays[i], hosts[i], _ = makeWakuRelay(t, log)
+		err := relays[i].Start(ctx)
+		require.NoError(t, err)
+	}
+
+	// Create peer manager instance with the first hosts
+	pm, eventBus := makePeerManagerWithEventBus(t, relays[0], &hosts[0])
+	pm.ctx = ctx
+	pm.RegisterWakuProtocol(relay.WakuRelayID_v200, relay.WakuRelayENRField)
+
+	// Connect host[0] with all other hosts to reach 4 connections
+	for i := 1; i < 5; i++ {
+		pm.host.Peerstore().AddAddrs(hosts[i].ID(), hosts[i].Addrs(), peerstore.PermanentAddrTTL)
+		err := pm.host.Connect(ctx, hosts[i].Peerstore().PeerInfo(hosts[i].ID()))
+		require.NoError(t, err)
+		err = pm.host.Peerstore().(wps.WakuPeerstore).SetDirection(hosts[i].ID(), network.DirOutbound)
+		require.NoError(t, err)
+
+	}
+
+	go pm.connectivityLoop(ctx)
+
+	time.Sleep(2 * time.Second)
+
+	if len(pm.host.Peerstore().(*wps.WakuPeerstoreImpl).PeersByPubSubTopic(pubSubTopic)) == 0 {
+		log.Info("No peers for the topic yet")
+	}
+
+	// Subscribe to Pubsub topic
+	for i := 0; i < 5; i++ {
+		_, err := relays[i].Subscribe(ctx, protocol.NewContentFilter(pubSubTopic))
+		require.NoError(t, err)
+	}
+
+	// peerEvt to find: relay.PEER_JOINED, relay.PEER_LEFT
+	peerEvt := relay.EvtPeerTopic{
+		PubsubTopic: pubSubTopic,
+		PeerID:      hosts[1].ID(),
+		State:       relay.PEER_JOINED,
+	}
+
+	emitter, err := eventBus.Emitter(new(relay.EvtPeerTopic))
```

Comment:
We don't need to emit these events as a peer joins a topic, the event get emitted from gossipsub layer

Reply:
I've simplified event handling test to focus on adding peers to topic and connectedness only. Getting to health  would require full stack and this could be tested better at interop level.